lm_eval/models/mamba_lm.py

```diff
@@ -0,0 +1,110 @@
+"""
+
+"""
+
+import torch
+
+import transformers
+from transformers import AutoTokenizer
+
+try:
+    from mamba_ssm.models.mixer_seq_simple import MambaLMHeadModel
+    from mamba_ssm.utils.hf import load_config_hf
+except ModuleNotFoundError:
+    pass
+
+from lm_eval import utils
+from lm_eval.api.model import LM
+from lm_eval.models.huggingface import HFLM
+from lm_eval.api.registry import register_model
+
+from typing import Literal, Optional, Union
+
+
+@register_model("mamba_ssm")
+class MambaLMWrapper(HFLM):
+    def __init__(
+        self,
+        pretrained="state-spaces/mamba-130m",
+        # mamba supports most default HFLM kwargs.
+        # however, it does not currently support advanced from_pretrained() kwargs
```

Comment:
Is "advanced" a classification of kwargs HF uses? If not, let's just list them all out.

Reply:
sgtm, will list all Mamba supported args (as HF kwargs are subject to change and expand in number)